### PR TITLE
Earn: show title of subscriber plans

### LINKED ISSUE #46157: Earn: Users cannot determine which users are on which plans if plans are identical.
If an Earn user creates multiple plans with the same details, they will not be able to determine which subscribers belong to which plans in our current UI. If the amount and frequency are the same for two plans, their subscribers are indistinguishable.

Screen Shot 2020-10-05 at 9 03 38 AM

See: p1601912260041700-slack-CDA5HLRD2

### PR DESCRIPTION
**Pull request by blackjackkent**
#### Changes proposed in this Pull Request

* This PR adds extra text to a user's list of active subscribers, indicating which subscription plan the subscription applies to. This makes it easier to distinguish between subscribers of different plans with the same price.

#### Testing instructions

* Sandbox the store and connect your account to Stripe via the steps described in PCYsg-lW4-p2, "Sandbox method"
* Navigate to `/earn/payments-plans/{siteId}` and add two new monthly payment plans with the same price but different descriptions.
* Create a post with two Premium Content blocks, each associated with one of the new payment plans.
* From another account, visit the post and subscribe to both payment plans.
    * Use one of the test credit card numbers from the above FG page.
* On your original account, navigate to `/earn/payments/{siteId}`. You should see both of your new subscriptions with the description attached to each, indicating which is which.
* Repeat the above with yearly instead of monthly payment plans.

Before:

image

After:

image

Fixes #46157

### CODE REVIEW

**Comment by blackjackkent** [comment]
Note: this seemed like the simplest solution, but @sixhours (or anyone else), feel free to add design suggestions! 😄

**Review by mmtr — COMMENTED** [review]
Great, much less confusing now 🙂 

I think we should also include the plan name to lifetime plans (one-time payments) as well: 

https://github.com/Automattic/wp-calypso/blob/ca6f6a3b5edc43e8a3ab73d3396248b88a412498/client/my-sites/earn/memberships/index.jsx#L401

Will defer to @sixhours on the design aspect, but it looks good to me 👍

File: client/my-sites/earn/memberships/index.jsx [hunk 1]
```diff
@@ -406,21 +406,23 @@ class MembershipsSection extends Component {
 			} );
 		} else if ( subscriber.plan.renew_interval === '1 year' ) {
 			return this.props.translate(
-				'Paying %(amount)s/year since %(formattedDate)s. Total of %(total)s.',
+				'Paying %(amount)s/year%(title)ssince %(formattedDate)s. Total of %(total)s.',
```
Comment (sixhours) [inline client/my-sites/earn/memberships/index.jsx:409]: Might be worth adding translator comments for these lines, especially where "title" could be ambiguous and the spacing changes depending on the variable.

File: client/my-sites/earn/memberships/index.jsx [hunk 2]
```diff
@@ -406,21 +406,23 @@ class MembershipsSection extends Component {
...
+						title: subscriber.plan.title ? ` (${ subscriber.plan.title }) ` : '',
 						formattedDate: this.props.moment( subscriber.start_date ).format( 'll' ),
 						total: formatCurrency( subscriber.all_time_total, subscriber.plan.currency ),
 					},
 				}
 			);
 		} else if ( subscriber.plan.renew_interval === '1 month' ) {
 			return this.props.translate(
-				'Paying %(amount)s/month since %(formattedDate)s. Total of %(total)s.',
+				'Paying %(amount)s/month%(title)ssince %(formattedDate)s. Total of %(total)s.',
```
Comment (sixhours) [inline client/my-sites/earn/memberships/index.jsx:421]: Same thing with translator comments to explain the variables.

**Comment by sixhours** [comment]
This looks good! :) Left a couple non-blocking comments.

**Pull request merged**